lib/BloqadeQMC/src/diagnostics/transition_matrix.jl

```diff
@@ -0,0 +1,92 @@
+using OnlineStats
+
+
+abstract type AbstractTransitionMatrix{T}; end
+
+# returns the transition matrix with row sums normalized to 1
+function LinearAlgebra.normalize(M::AbstractTransitionMatrix{T}) where T
+    rowsums = sum(M.m, dims=2)
+
+    normalized = similar(M.m, float(T))
+    for i in eachindex(rowsums)
+        if !iszero(rowsums[i])
+            @. normalized[i, :] = M.m[i, :] / rowsums[i]
+        else
+            @. normalized[i, :] = float(M.m[i, :])
+        end
+    end
+    return normalized
+end
+
+
+struct NoTransitionMatrix <: AbstractTransitionMatrix{Int}; end
+@inline OnlineStats.fit!(M::NoTransitionMatrix, _, _) = M
```

Comment:
i have no idea why i was importing the OnlineStats function here, I'll change this later